client/my-sites/stats/features/modules/stats-top-posts/stats-top-posts.tsx

```diff
@@ -49,7 +52,18 @@ const StatsTopPosts: React.FC< StatsTopPostsProps > = ( {
 		<>
 			{ /* This will be replaced with ghost loaders, fallback to the current implementation until then. */ }
 			{ requesting && <StatsModulePlaceholder isLoading={ requesting } /> }
-			{ ( ! data || ! data?.length ) && (
+			{ /* TODO: consider supressing <StatsModule /> empty state */ }
+			{ ( data && !! data.length ) || shouldGateStatsTopPosts ? (
```

Comment:
Not related to the PR and the PR could be a fix: I suspect the upsells are broken with the introduction of 'StatsCard'. 🤔

Reply:
I am pretty sure about the situation you described here. However, the upsell here is working within the `StatsModule`.

Reply:
But indeed, we need to tackle other modules wrapped with new empty state structures recently, e.g., [StatsReferrers](https://github.com/Automattic/wp-calypso/pull/92153) and [StatsCountries](https://github.com/Automattic/wp-calypso/pull/92180).

Reply:
I think you are right @dognose24. We need to get `StatsCard` to work with upsell overlay before releasing or at least make sure the upsell overlay is not broken in these cases.

cc @grzegorz-cp